app/models/api_key.rb

```diff
@@ -0,0 +1,11 @@
+class ApiKey < ApplicationRecord
+  belongs_to :user
+  validates :user, :name, :hashed_key, :expires_at, presence: true
+  before_validation :set_expires_at
```

Comment:
Should updates to `ApiKey`s call `set_expires_at`?

Reply:
ApiKey doesn't expose update to public API. Users can only create and destroy keys.